pkg/controller/controller.go

```diff
@@ -689,6 +700,20 @@ func (p *csiProvisioner) Provision(ctx context.Context, options controller.Provi
 		}
 	}
 
+	pvReadOnly := false
+	volCaps := req.GetVolumeCapabilities()
+	// if the request only has one accessmode and if its ROX, set readonly to true
+	if len(volCaps) == 1 {
+		if volCaps[0].GetAccessMode().GetMode() == csi.VolumeCapability_AccessMode_SINGLE_NODE_READER_ONLY || volCaps[0].GetAccessMode().GetMode() == csi.VolumeCapability_AccessMode_MULTI_NODE_READER_ONLY {
+			if (scReadOnlyKeyFound == 1) && scReadOnlyEnabled {
```

Comment:
Can we just check scReadOnlyEnabled?

Also, is this something that we think should be enabled on a per storageclass basis, or on a driver basis? If driver, then a flag to external provisioner would be better.